### PR TITLE
Set precision and recall to 0 when undefined

### LINKED ISSUE #328: Question about computation of precision and recall
Hi Nicholas,
I was looking at the computation of precision and recall 

        # Precision@K: Proportion of recommended items that are relevant
        precisions[uid] = n_rel_and_rec_k / n_rec_k if n_rec_k != 0 else 1

        # Recall@K: Proportion of relevant items that are recommended
        recalls[uid] = n_rel_and_rec_k / n_rel if n_rel != 0 else 1

and I was wondering why we assign 1 if n_rec_k  is 0 which means that there are not good recommendations. Isn't it wrong? This will increase the final mean over all the users.

Thank you,
Noemi

### PR DESCRIPTION
**Pull request by ashtou**

### CODE REVIEW

**Comment by NicolasHug** [comment]
Thanks @ashtou . Could you please make sure the lines don't exceed 79 characters?

**Comment by ashtou** [comment]
Is that what you meant? I can also squash the commits and force push it, if need be

File: examples/precision_recall_at_k.py [hunk 1]
```diff
@@ -37,10 +37,12 @@ def precision_recall_at_k(predictions, k=10, threshold=3.5):
                               for (est, true_r) in user_ratings[:k])
 
         # Precision@K: Proportion of recommended items that are relevant
-        precisions[uid] = n_rel_and_rec_k / n_rec_k if n_rec_k != 0 else 1
+        # When n_rec_k is 0, Precision is undefined. As an alternative, it can be set to 0
```
Comment (NicolasHug) [inline examples/precision_recall_at_k.py:40]:
```suggestion
        # When n_rec_k is 0, Precision is undefined. We here set it to 0.

```

File: examples/precision_recall_at_k.py [hunk 2]
```diff
@@ -37,10 +37,12 @@ def precision_recall_at_k(predictions, k=10, threshold=3.5):
                               for (est, true_r) in user_ratings[:k])
 
         # Precision@K: Proportion of recommended items that are relevant
-        precisions[uid] = n_rel_and_rec_k / n_rec_k if n_rec_k != 0 else 1
+        # When n_rec_k is 0, Precision is undefined. As an alternative, it can be set to 0
+        precisions[uid] = n_rel_and_rec_k / n_rec_k if n_rec_k != 0 else 0
 
         # Recall@K: Proportion of relevant items that are recommended
-        recalls[uid] = n_rel_and_rec_k / n_rel if n_rel != 0 else 1
+        # When n_rel is 0, Recall is undefined. As an alternative, it can be set to 0
```
Comment (NicolasHug) [inline examples/precision_recall_at_k.py:44]:
```suggestion
        # When n_rel is 0, Recall is undefined. We here set it to 0.

```

**Comment by NicolasHug** [comment]
Thanks @ashtou !

**Pull request merged**